source/distributed/shared_tria.cc

```diff
@@ -134,6 +134,16 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      dealii::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
```

Comment:
there is `MPI_Comm_free` in destructor of `p::Tria`, but are right, it's not freed before assignment in `p::Tria::copy_triangulation()`. @bangerth @tjhei is it a bug?